### DISCUSSION TITLE
Potential Undefined Attributes in pixi.lock for purl Generation

### DISCUSSION THREAD

**Discussion by aryan-rajoria** [prefix-dev/pixi#1932] [Q&A] [answered]
I'm working on generating Package URL (purl) representations for packages listed within a `pixi.lock` file. I'm using the following template:
```
const purlTemplate = `pkg:${packageData["kind"]}/${packageData["name"]}@${packageData["version"]}-${packageData["build"]}?os=${packageData["subdir"]}`;
```
However, I'm concerned that some of the attributes I'm referencing (like `build` or `subdir`) might be undefined in certain cases. Could someone familiar with the structure of `pixi.lock` files confirm which attributes are always present? Any insights or suggestions to make my PURL generation more robust would be greatly appreciated.

**Comment by baszalmstra** [comment, marked as answer]
For conda packages the build and subdir cannot be optional. See: https://docs.rs/rattler_conda_types/0.27.2/rattler_conda_types/struct.PackageRecord.html which is  stored in the lock file. 

for pypi packages there is only name and version. See: https://docs.rs/rattler_lock/latest/rattler_lock/struct.PypiPackageData.html

I think you need to handle those cases seperately
